libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/Clock.kt

```diff
@@ -0,0 +1,11 @@
+package com.mapbox.navigation.navigator.internal
+
+import android.os.SystemClock
+
+interface Clock {
```

Comment:
Can we delete [Time](https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/Time.kt)?

Or do we need multiple implementations right now?

Reply:
`Time` is exposed in `libnavigation-util` and it's used in various other modules. `libnavigator` doesn't depend on `libnavigation-util` so as it's a small interface that is implementing the `elapsedRealtimeNanos` (not available in `Time` implementation) decided to create `Clock`. I also considered including `Time` in `libnavigation-base` but it doesn't really belong to that module so that was another reason that made me go with current approach.

Reply:
Ok, can we make this a class. It'll be easy for testing if we construct it when we need to use it. Some examples. Otherwise if we want to mock the object, we need to do it statically

``` kotlin
class Clock {
   
    fun elapsedRealtimeNanos(): Long = monoTimeNanos()

    companion object {
        private fun monoTimeNanos(): Long = System.elapsedRealtimeNanos()
    }
}
```
``` kotlin
class Example(clock: Clock = Clock()) {
    fun doSomething() {
        // use clock
    }

} 
```

``` kotlin
class ExampleTest {

    @Test
    fun exampleTest() {
         val clock: Clock = mockk {
               every { elapsedRealtimeNanos() } returns 10000
         }
    }
}
```